### PR TITLE
Adds Log4j 1.x global threshold

### PR DESCRIPTION
**Pull request by ppkarwasz**
The `log4j.threshold` configuration key was not translated into its Log4j 2.x equivalent global filter.

### CODE REVIEW

File: log4j-1.2-api/src/main/java/org/apache/log4j/config/Log4j1ConfigurationParser.java [hunk 1]
```diff
@@ -103,6 +107,18 @@
             if (Boolean.parseBoolean(debugValue)) {
                 builder.setStatusLevel(Level.DEBUG);
             }
+            // if log4j.reset=true then reset hierarchy
+            final String reset = getLog4jValue("reset");
+            if (reset != null && OptionConverter.toBoolean(reset, false)) {
```
Comment (garydgregory) [inline log4j-1.2-api/src/main/java/org/apache/log4j/config/Log4j1ConfigurationParser.java:112]: I do not see this reset setting tested in this PR. Did I miss it? Also, elsewhere we call the reset API from a doConfigure() method, it seems odd to call it here. Is it then called twice or should this call be moved to the site when the configuration is actually applied? @rgoers WDYT?

Comment (ppkarwasz) [reply log4j-1.2-api/src/main/java/org/apache/log4j/config/Log4j1ConfigurationParser.java:112]: Yes, you are right. I copy-pasted the "reset" part from the `PropertiesConfiguration`, without checking exactly what it does. I'll remove it for now (besides the `XmlConfiguration` does not recognize this option).

Comment (ppkarwasz) [reply log4j-1.2-api/src/main/java/org/apache/log4j/config/Log4j1ConfigurationParser.java]: Looking at the Log4j 1.2 code, neither `Log4j1ConfigurationParser` nor any `Configuration` class should check the _"log4j.reset"_ key. This should be done in `PropertyConfigurator#configure` to decide whether to replace the old configuration or to make a `CompositeConfiguration` out of them.

Looking at [this SO question](https://stackoverflow.com/q/71087714/11748454) I have the impression that calling `PropertyConfigurator#configure` multiple times without resetting the old config, was a common practice.

**Comment by garydgregory** [comment]
@ppkarwasz 
Thank you for your PR. 
If you are looking for more fixes to investigate, please have a look at the disabled tests in `org.apache.log4j.xml.DOMTestCase`.

**Pull request merged**